seriouslysean/branch-history-force-push-reproduction
# Feature/main at step 7

### Ticket:
https://urbnit.atlassian.net/browse/MBH-?????

### Testing Steps:
* First pass of this branch was the way I originally added a new commit to modify the codeowners. I added the change, then pulled main directly which made a merge commit. That ended up giving something like this:
![image](https://user-images.githubusercontent.com/414475/164542343-6e029800-edf1-4fed-a1e1-d6450d0df279.png)
* After doing that, to keep the git history clean I opted to `git reset HEAD^` and remove the leading git commit (the merge commit), thereby removing it, which left me with this: 
![image](https://user-images.githubusercontent.com/414475/164542525-e1080d14-0328-4710-a7cf-0e8f2428bccb.png)
* That had the unintended consequence of removing the history included in that merge (which fast forwarded my changes in one commit by the amount I was behind by). This is what caused some of the PRs pointed to main to show that commits were missing, https://github.com/seriouslysean/branch-history-force-push-reproduction/pull/9
![Screen Shot 2022-04-21 at 3 59 00 PM](https://user-images.githubusercontent.com/414475/164542939-360350f4-dd7e-410a-986f-28ee9d051a9b.png)
* In order to fix this issue, I had to look up the last good commit (in the first case Jenkins, but in this case I can just reference the clean `main` branch I left intact, 7f488426c2429de65495b5a3efe92c4e223cd046.
![image](https://user-images.githubusercontent.com/414475/164543157-b9c76b5b-cf4e-46e3-9a33-6a935be875f0.png)
* Once that sha1 was found, I was able to reset the `main` branch once more with the last clean sha which again showed no difference between what we had before and the new sha1 I pushed up:
![image](https://user-images.githubusercontent.com/414475/164543373-0e5ffa64-7c08-4ffa-a65b-3306f82c8764.png)
![image](https://user-images.githubusercontent.com/414475/164543405-18a60083-bee7-4f2a-91b8-642e223766a6.png)
* The last step was checking the affected PRs to make sure they were ok. The PR was still showing the out of date main reference, so I had to rebase that branches work on the clean main sha1, thus fixing the last remaining issue.
![image](https://user-images.githubusercontent.com/414475/164543682-b0a35eb2-75a1-4b76-bdaa-690d6cc7c4f0.png)
![image](https://user-images.githubusercontent.com/414475/164544184-92bee223-6c85-4e4d-b100-031fbbf06532.png)
